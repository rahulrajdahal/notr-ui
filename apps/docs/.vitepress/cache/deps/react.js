import {
  require_react
} from "./chunk-UUF3TEGA.js";
import "./chunk-5WRI5ZAA.js";
export default require_react();
//# sourceMappingURL=react.js.map
