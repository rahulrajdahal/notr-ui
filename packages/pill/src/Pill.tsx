import { css } from "@emotion/react";
import { colors } from "@rahulrajdahal/notr-core";
import React from "react";

const baseStyles = css`
  display: flex;
  height: 2.5rem;
  width: 100%;
  max-width: 10.25rem;
  align-items: center;
  justify-content: center;
  padding: 0 0.25rem;
  gap: 0.44rem;
  border-radius: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 0.875rem;
`;

export interface PillProps extends React.HTMLAttributes<HTMLSpanElement> {
  /** Prefix icon component */
  icon?: React.ReactNode;
  /** Number of items */
  count: number;
  /** Text content of Pill */
  text: string;
  /** Background color of Pill */
  background?: string;
  /** text color of Pill */
  color?: string;
}

const Pill = React.forwardRef<HTMLSpanElement, PillProps>(
  (
    {
      text,
      icon,
      count,
      background = colors.green[400],
      color = colors.green[600],
      ...props
    },
    ref
  ) => {
    return (
      <span
        ref={ref}
        style={
          {
            backgroundColor: background,
            color,
            "--color-green-400": colors.green[400],
            "--color-green-600": colors.green[600],
          } as React.CSSProperties
        }
        css={baseStyles}
        {...props}
      >
        {icon} {count} task
        {count > 1 ? "s" : ""} {text}
      </span>
    );
  }
);

Pill.displayName = "Pill";

export default Pill;
