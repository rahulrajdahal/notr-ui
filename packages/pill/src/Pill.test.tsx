import { createSerializer } from "@emotion/jest";
import { CheckSquare } from "@meistericons/react";
import { colors } from "@rahulrajdahal/notr-core";
import { render, screen } from "@testing-library/react";
import Pill from "./Pill";

describe("Button Component", () => {
  it("renders with default props and styles", () => {
    render(<Pill data-testid="notr-pill" text="Completed" count={3} />);
    const button = screen.getByTestId("notr-pill");
    expect(button).toBeInTheDocument();
    expect(button).toHaveStyle(`--color-green-600: ${colors.green[600]}`);
    expect(button).toHaveStyle(`--color-green-400: ${colors.green[400]}`);
    expect(button).toHaveStyle(`background-color: ${colors.green[400]}`);
    expect(button).toHaveStyle(`color: ${colors.green[600]}`);
    expect(button).toHaveStyle("display: flex");
    expect(button).toHaveStyle("height: 2.5rem");
    expect(button).toHaveStyle("border-radius: 0.75rem");
    expect(button).toHaveStyle("max-width: 10.25rem");
    expect(button).toHaveStyle("align-items: center");
    expect(button).toHaveStyle("justify-content: center");
    expect(button).toHaveStyle("font-weight: 500");
    expect(button).toHaveStyle("padding: 0 0.25rem");
    expect(button).toHaveStyle("gap: 0.44rem");
    expect(button).toHaveStyle("font-size: 0.875rem");
    expect(button).toHaveStyle("line-height: 0.875rem");

    expect(button.innerHTML).toBe(" 3 tasks Completed");

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();
  });

  it("renders with icon props and styles", () => {
    render(
      <Pill
        data-testid="notr-pill"
        icon={<CheckSquare />}
        text="Completed"
        count={3}
      />
    );
    const button = screen.getByTestId("notr-pill");
    expect(button).toBeInTheDocument();

    expect.addSnapshotSerializer(createSerializer());
    expect(button).toMatchSnapshot();
  });
});
