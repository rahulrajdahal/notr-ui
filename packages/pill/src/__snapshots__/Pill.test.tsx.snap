// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > renders with default props and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  width: 100%;
  max-width: 10.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 0.25rem;
  gap: 0.44rem;
  border-radius: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 0.875rem;
}

<span
  class="emotion-0"
  data-testid="notr-pill"
  style="background-color: rgba(9, 196, 151, 0.2); color: rgb(9, 196, 151); --color-green-400: #09c49733; --color-green-600: #09c497;"
>
   
  3
   task
  s
   
  Completed
</span>
`;

exports[`Button Component > renders with icon props and styles 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  width: 100%;
  max-width: 10.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 0.25rem;
  gap: 0.44rem;
  border-radius: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 0.875rem;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  width: 100%;
  max-width: 10.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0 0.25rem;
  gap: 0.44rem;
  border-radius: 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 0.875rem;
}

<span
  class="emotion-0"
  data-testid="notr-pill"
  style="background-color: rgba(9, 196, 151, 0.2); color: rgb(9, 196, 151); --color-green-400: #09c49733; --color-green-600: #09c497;"
>
  <svg
    class="mni mni-check-square"
    color="currentColor"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M3.75 9c0-2.886 2.364-5.25 5.25-5.25h6c2.886 0 5.25 2.364 5.25 5.25v6c0 2.886-2.364 5.25-5.25 5.25H9c-2.886 0-5.25-2.364-5.25-5.25zM9 2.25C5.286 2.25 2.25 5.286 2.25 9v6c0 3.714 3.036 6.75 6.75 6.75h6c3.714 0 6.75-3.036 6.75-6.75V9c0-3.714-3.036-6.75-6.75-6.75zm7.425 7.585a.75.75 0 0 0-1.05-1.07l-5.038 4.943-1.743-2.265a.75.75 0 1 0-1.188.914l2.03 2.639.034.034c.181.182.46.37.83.37.367 0 .645-.186.827-.367z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
   
  3
   task
  s
   
  Completed
</span>
`;
